### PR TITLE
Allow to use spark-internal as resource for batch job

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Allow to use spark-internal as resource for batch job.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master and 1.6.2

**Comment by turboFei** [comment]
thanks, merged to master and 1.6.2

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala [hunk 1]
```diff
@@ -190,6 +190,7 @@ object SparkProcessBuilder {
   final val APP_KEY = "spark.app.name"
   final val TAG_KEY = "spark.yarn.tags"
   final val MASTER_KEY = "spark.master"
+  final val INTERNAL_RESOURCE = "spark-internal"
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala:193]: shall we docs the special flag ?
